core/config/project_settings.cpp

```diff
@@ -54,6 +56,73 @@ String ProjectSettings::get_resource_path() const {
 
 const String ProjectSettings::IMPORTED_FILES_PATH("res://.godot/imported");
 
+// Returns the features that a project must have when opened with this build of Godot.
+// This is used by the project manager to provide the initial_settings for config/features.
+const PackedStringArray ProjectSettings::get_required_features() {
+	PackedStringArray features = PackedStringArray();
+	features.append(VERSION_BRANCH);
+#ifdef REAL_T_IS_DOUBLE
+	features.append("Double-precision");
+#else
+	features.append("Single-precision");
+#endif
+	return features;
+}
+
+// Returns the features supported by this build of Godot. Includes all required features.
+const PackedStringArray ProjectSettings::_get_supported_features() {
+	PackedStringArray features = get_required_features();
+#ifdef MODULE_MONO_ENABLED
+	features.append("C#");
+#endif
+	// Allow pinning to a specific patch number or build type by marking
+	// them as supported. They're only used if the user adds them manually.
+	features.append(VERSION_BRANCH "." _MKSTR(VERSION_PATCH));
+	features.append(VERSION_FULL_CONFIG);
+	features.append(VERSION_FULL_BUILD);
+	// For now, assume Vulkan is always supported.
+	// This should be removed if it's possible to build the editor without Vulkan.
+	features.append("Vulkan 1.0 Clustered");
+	features.append("Vulkan 1.0 Mobile");
```

Comment:
It's possible for Godot to use extensions from Vulkan 1.1 and 1.2 if supported. Godot will do so automatically as needed (e.g. for multiview). Should the version 1.0 still be listed in this case?

Reply:
This is a good question, which I'm not sure of the answer to. I included the minor version just in case, but if Vulkan extensions are used automatically yet not required, perhaps this isn't necessary. Is there a chance that Godot would require Vulkan 1.x versions in the future? If so, "1.0" makes sense, if not, it should be "Vulkan 1" (and not just "Vulkan" in the event that Vulkan 2+ is significantly different).

Reply:
> Is there a chance that Godot would require Vulkan 1.x versions in the future?

I don't know for sure, but it's unlikely because Vulkan 1.1 support is still poor on mobile platforms right now.